src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -3573,7 +3583,12 @@ public Socket EndAccept(IAsyncResult asyncResult)
         }
         private Socket EndAcceptCommon(out byte[]? buffer, out int bytesTransferred, IAsyncResult asyncResult)
         {
-            ThrowIfDisposed();
+            if (Disposed)
+            {
+                if (SocketsTelemetry.Log.IsEnabled()) SocketsTelemetry.Log.AfterAccept(SocketError.OperationAborted);
```

Comment:
Shouldn't we log this in the completion callbacks instead?

Reply:
We aren't flowing the ExecutionContext to the completion callbacks.
We only restore it when calling the user-provided callback.